File: javascript/hamming/hamming.js
var Hamming = function(){};

Hamming.prototype.compute = function(dna1, dna2){
	if(dna1.length !== dna2.length){
		throw 'DNA strands must be of equal length.';
	}
	var ham = 0;
	for(var i = 0; i < dna1.length; i++){
		if(dna1[i] !== dna2[i]){
			ham++;
		}
	}
return ham;
}

module.exports = Hamming;

File: java/hamming/src/main/java/Hamming.java
class Hamming {

    String left;
    String right;

    Hamming(String leftStrand, String rightStrand) {
        if (!leftStrand.isEmpty() && rightStrand.isEmpty()) throw new IllegalArgumentException("right strand must not be empty.");

        if (!rightStrand.isEmpty() && leftStrand.isEmpty()) throw new IllegalArgumentException("left strand must not be empty.");

        if (leftStrand.length() != rightStrand.length()) throw new IllegalArgumentException("leftStrand and rightStrand must be of equal length.");

        left = leftStrand;
        right = rightStrand;
    }

    int getHammingDistance() {
        int ham = 0;
        for (int i = 0; i < right.length(); i++){
            if (right.charAt(i) != left.charAt(i)){
                ham++;
            }
        }
        return ham;
    }

}
